graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultInputObjectMapper.kt

```diff
@@ -67,6 +69,18 @@ class DefaultInputObjectMapper(customInputObjectMapper: InputObjectMapper? = nul
             if (KotlinDetector.isKotlinType(targetType.type)) {
                 return mapper.mapToKotlinObject(sourceMap, targetType.type.kotlin)
             }
+
+            if (targetType.resolvableType.hasGenerics()) {
```

Comment:
Should this check specifically for `Optional` instead of any type with generics?